### PR TITLE
Serialized key function

### LINKED ISSUE #7: [RFC] - Allow property name to take a function in addition to a fixed string
This will allow for programmatic re-naming of property keys such as automatic case conversion.

### PR DESCRIPTION
**Pull request by PizzaCatKing**
## Fixes

fixes #7 

## Description

Adds the ability to use a property key transform function as a serialized key

## Proposed changes in this PR

- Adds ability to pass a function that dynamically transforms the property key into a serialized key

## Things to look at

- [ ] Test coverage
- [ ] Code Style
- [ ] Documentation (READMEs etc)

### CODE REVIEW

File: serialize_property_test.ts [hunk 1]
```diff
@@ -380,3 +380,72 @@ test({
...
   },
 });
+
+test({
+  name: "Serialize key function",
+  fn() {
+    class Test1 extends Serializable {
+      @SerializeProperty(
+        (
+          (propertyName) => "_" + (propertyName as string)
```
Comment (PizzaCatKing) [inline serialize_property_test.ts]: This cast is kind of dirty, I think a better way of handling it would be to throw on `Symbol`.

Comment (hardy925) [reply serialize_property_test.ts]:
```suggestion
          (propertyName) => `_${String(propertyName)}`
```

I believe this calls `.toString` on Symbol

**Comment by hardy925** [comment]
@PizzaCatKing thoughts on this pr?

**Comment by hardy925** [comment]
hey @PizzaCatKing what is left for this PR? do we need more tests or docs?

File: serialize_property.ts [hunk 2]
```diff
@@ -71,12 +82,19 @@ export function SerializeProperty(
...
+          fromJsonStrategy: decoratorArguments.fromJsonStrategy,
+          toJsonStrategy: decoratorArguments.toJsonStrategy,
+        };
+      } else {
+        decoratorArgumentOptions = {
+          // we can always define serializedKey as decoratorArguments.serializedKey will override this
+          serializedKey: (target as any).tsTransformKey(propertyName),
+          ...decoratorArguments,
+        };
+      }
```
Comment (hardy925) [inline serialize_property.ts]: what if we abstract setting `decoratorArgumentOptions` to its own function?

**Review by ms1111 — APPROVED** [review]
I figure this layers on top of #66, so #66 lets you specify a class-level function for serialized keys and this one lets you specify a property-level function. The order of precedence doesn't change: you can specify a string or function at the property level, and if you don't, the class-level tsTransformKey() will be used.

**Review by hardy613 — COMMENTED** [review]
I don't think we'll need `ToSerializedKeyStrategy` for node right, 

this is awesome though.

**Review by hardy925 — APPROVED** [review]
LGTM! thank you

**Pull request merged**